scalafmt-core/shared/src/main/scala/org/scalafmt/util/TreeOps.scala

```diff
@@ -676,7 +679,13 @@ object TreeOps {
     }
 
   // Redundant {} block around case statements
-  def isCaseBodyABlock(ft: FormatToken, caseStat: Case): Boolean =
-    ft.right.is[Token.LeftBrace] && (caseStat.body eq ft.meta.rightOwner)
+  def isCaseBodyABlock(ft: FormatToken, caseStat: Tree): Boolean = {
+    val body = caseStat match {
+      case tc: TypeCase => tc.body
+      case c: Case => c.body
+    }
+    ft.right.is[Token.LeftBrace] && (body eq ft.meta.rightOwner) &&
+    (caseStat.is[Case] || caseStat.is[TypeCase])
```

Comment:
it's too late to check that parameter is one of them. perhaps you should set body to null in the match above instead.

Reply:
Added an option and I check it here.